pkg/graph/resolvers/model_plan.go

```diff
@@ -9,12 +9,21 @@ import (
 )
 
 // ModelPlanCreate implements resolver logic to create a model plan
-func ModelPlanCreate(logger *zap.Logger, plan *models.ModelPlan, store *storage.Store) (*models.ModelPlan, error) {
+func ModelPlanCreate(logger *zap.Logger, plan *models.ModelPlan, store *storage.Store, principalInfo *models.UserInfo) (*models.ModelPlan, error) {
 
 	createdPlan, err := store.ModelPlanCreate(logger, plan)
```

Comment:
I wonder if it makes sense to have a `store` method that could wrap
1. Create Model Plan
2. Create Plan Collaborator

in a [sqlx transaction](https://jmoiron.github.io/sqlx/#transactions) so that they're done atomically 🤔 

This would have the benefit of making the error handling in this function simpler, as you wouldn't have to handle the `err` of `ModelPlanCreate` and the `err` of `CreatePlanCollaborator` individually.

Might not be worth the lift right now, but a thought to consider any time we have multiple unique SQL statements to make as part of a single resolver.

Reply:
Hmm  🤔, not a bad thought

The only thing I don't like about that is you would have a store file for model_plan, that would also be creating database records for plan collaborator. That could be fine.

I guess the other question is, would we want to not create the model plan if the plan collaborator fails to be inserted? If so, the sqlx transaction would make sense so we could roll it back

Reply:
That was what I was thinking (roll it back if we can't make the collaborator) - because if you fail to make the collaborator, you're left in a spot where nobody can access the plan since there's 0 collaborators.

I don't think we need to solution this right now, but can you leave a `//TODO` comment in there clarifiying it's something we should be aware of?

Reply:
Also - I agree with you on that - the nice part about Go packages is we can have a file for methods like this while keeping it in the same package

```go
package store
```

in a file like `pkg/store/model_plan_collab.go` - though I'm not sold on this right now lol

Reply:
that's a good thought, and it isn't even necessarily a problem to have the action happen in both stores. Another option which I kind of like is to roll it into one SQL statement. I don't know if there is ever a situation where we want to make a model plan without a collaborator. The main thing I don't like about that is you have multiple entry points to write to the table. It can just start to get messy, thought it would be the most efficient method

Reply:
added a comment to address this for now